src/libre/dialect/pcre/lexer.lx

```diff
@@ -39,8 +39,27 @@
 '\t' -> $esc;
 '\v' -> $esc;
 
-'\'  /[0-7]+/     -> $oct;
-'\x' /[0-9a-f]+/i -> $hex;
+# perl/unix-ish escapes
+'\a' -> $esc;
+'\c' /./ -> $control;
+
+# perl character class escapes
+'\d' -> $class_digit;
+'\D' -> $class_ndigit;
+'\w' -> $class_word;
+'\W' -> $class_nword;
+'\s' -> $class_space;
+'\S' -> $class_nspace;
+
+class = '\' /[dDwWsS]/;
+
+'\'  /[0-7]{1,3}/      -> $oct;
+'\o{' /[0-7]+/i '}'    -> $oct;
+'\x' /[0-9a-f]{1,2}/i  -> $hex;
+'\x{' /[0-9a-f]+/i '}' -> $hex;
+
+'\' /./ - ($esc | $control | $oct | $hex | '\Q' | class) -> $noesc;
+'\' -> $invalid;
 
```

Comment:
Rather than
```
 	'i' -> $flag__insensitive;
 	/[a-z]/ - /[i]/ -> $flag__unknown;
```
you can write
```
 	'i' -> $flag__insensitive;
 	/[a-z]/ - $flag__insensitive -> $flag__unknown;
```